### ISSUE TITLE
Errors when deleting clients

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#525]
**Reported by valleegr on 25 Feb 2009 22:26 UTC**
```
=============================================================================
== oscar_wizard: Starting...
=============================================================================


=============================================================================
== Running step 5 of the OSCAR wizard: Define OSCAR clients
=============================================================================

Argument "2.121_08" isn't numeric in subroutine entry at /usr/share/perl5/MLDBM/Serializer/Data/Dumper.pm line 5
>> Turning off generic services
>> Executing post_clients phase
Populating /root/.ssh/known_hosts with defined clients' host keys
>> Executing post_install phase
ERROR: Couldn't push /etc/hosts to all nodes at /usr/bin/post_install line 80
ERROR: Some of the post install scripts failed, please check your logs for more info at /usr/bin/post_install line 93
post_install phase failed. at /usr/lib/perl5/Tk.pm line 247
>> Re-starting generic services
>> Re-starting server services
>> Updating C3 configuration file
C3 configuration file update phase failed. at /usr/lib/perl5/Tk.pm line 247
>> Re-starting client services on remaining nodes
Use of uninitialized value in string eq at /usr/share/perl5/OSCAR/DelNode.pm line 254.
Running mkdhcpconf
Restarting dhcpd
 * Stopping DHCP server dhcpd3       ESC[^MESC[74G[ESC[31mfailESC[39;49m](80G)
 * Starting DHCP server dhcpd3       ESC[^MESC[74G[ESC[31mfailESC[39;49m](80G)
Tk::Error: Failed to run /etc/init.d/dhcp3-server restart at /usr/lib/perl5/Tk.pm line 247
 Carp::croak at /usr/share/perl/5.8/Carp.pm line 102
 OSCAR::DelNode::delnodes at /usr/share/perl5/OSCAR/DelNode.pm line 266
 Tk callback for .toplevel.button1
 Tk::__ANON__ at /usr/lib/perl5/Tk.pm line 247
 Tk::Button::butUp at /usr/lib/perl5/Tk/Button.pm line 111
 <ButtonRelease-1>
 (command bound to event)
```

It seems that the problem come from the fact dhcpd was not running. We should check the status of the service before to try to restart it: if the service is running, we "restart"; otherwise, we "start".

**Comment by dikim33** [comment]
**Comment by valleegr on 21 Sep 2010 16:04 UTC**
The C3 error should not happen again since the C3 configuration file is updated when deleting nodes (upcoming C3 release).

**Comment by dikim33** [comment]
**Comment by valleegr on 21 Sep 2010 18:45 UTC**
(In [9299]) This check-in fixes #526 - Improve system sanity when deleting clients: DHCP is now correctly handled.
